tests/bpftrace.cpp

```diff
@@ -111,7 +111,8 @@ void check_special_probe(Probe &p, const std::string &attach_point, const std::s
 
 TEST(bpftrace, add_begin_probe)
 {
-  ast::AttachPoint a("BEGIN");
+  ast::AttachPoint a("");
```

Comment:
Is it possible to just do this instead of the separate assignment? Just to keep the tests a bit cleaner
```suggestion
  ast::AttachPoint a("BEGIN");
```

Reply:
I looked into the options and here is a summary of my thoughts:
* I would prefer to avoid parsing outside of the parsing unit tests. The reason being unit tests should be as orthogonal to each other as possible.
* I would also prefer to avoid all the overloaded constructors we had before (b/c it was hard to reason about which overload would get selected)
* Designated initializers looked pretty good, but there's too many language restrictions on them in C++ so I don't think it's possible to use it here (no user declared or inherited constructors allowed)

It seems like this is the best we can do unless you have other thoughts.